components/sharing-dialog/src/sharing-dialog.js

```diff
@@ -159,7 +159,13 @@ export const SharingDialog = ({
 
     return (
         <FetchingContext.Provider value={mutating || fetching}>
-            <Modal onClose={onClose} name={object.displayName || object.name}>
+            <Modal
+                onClose={onClose}
+                name={
+                    (object.displayName !== object.id && object.displayName) ||
```

Comment:
I think this can be simplified but just using `displayName`.
`displayName` should always have a value, either the translated version or the original version as default.
But we should confirm first.

Reply:
From the perspective of the fix, both `displayName` and `name` use the `id`, as seen e.g. here https://debug.dhis2.org/dev/api/39/sharing?type=interpretation&id=HOFzdanjMHE. But I see your point that there might not be a case where `displayName` is empty but `name` is set, that's what I've seen in my testing, however I can't confirm that to always be true.